tooling/compare-builds.sh

```diff
@@ -0,0 +1,604 @@
+  local TAG_NON_URL="${TAG_NON_URL}"
+  local TAG="${TAG}"
+  local ADOPTIUM_ARCH="${ADOPTIUM_ARCH}"
+  local OS="${OS}"
+  local ADOPTIUM_JSON="${ADOPTIUM_JSON}"
+  curl -X 'GET' \
+    "https://api.adoptium.net/v3/assets/release_name/eclipse/${TAG}?architecture=${ADOPTIUM_ARCH}&heap_size=normal&image_type=jdk&project=jdk&os=${OS}" \
+    -H 'accept: application/json' > $ADOPTIUM_JSON
+  if [ ! -s $ADOPTIUM_JSON ]; then
+    set +x
+	  echo "AQA api returned nothing! See avaialble versions below"
+      getSurroundingReleases "${ADOPTIUM_ARCH}" "${OS}" '"release_name"' "${TAG_NON_URL}" "${TAG_NON_URL}"
+	  echo "AQA api returned nothing! See avaialble versions above"
+    set -x
+    exit 1
+  fi
+  cat ${ADOPTIUM_JSON}
+  LAST_ADOPTIUM_JSON=${ADOPTIUM_JSON}
+  ls > before
+  curl -OLJSks "https://api.adoptium.net/v3/binary/version/${TAG}/${OS}/${ADOPTIUM_ARCH}/jdk/hotspot/normal/eclipse?project=jdk"
+  ls > after
+  ORIG_LAST_DOWNLOADED_FILE=$(getDiff)
+  local ORIG_LAST_DOWNLOADED_FILE="${ORIG_LAST_DOWNLOADED_FILE}"
+  LAST_DOWNLOADED_FILE="ojdk$(date +%s)_${ORIG_LAST_DOWNLOADED_FILE}"
+  mv "${ORIG_LAST_DOWNLOADED_FILE}" "${LAST_DOWNLOADED_FILE}"
+  if [ ! -e "${LAST_DOWNLOADED_FILE}" ]; then
+    set +x
+	  echo "AQA api downloaded nothing! See avaialble versions below"
+      getSurroundingReleases "${ADOPTIUM_ARCH}" "${OS}" '"link"' "${TAG_NON_URL}" "${TAG_NON_URL}"
+	  echo "AQA api downloaded nothing! See avaialble versions above"
+    set -x
+    exit 1
+  fi
+}
+
+function copyJdk () {
+  local jdk_source="${1}"
+  local jdk_name="${2}"
+  echo "${jdk_source} is directory, will be copied AS ${jdk_name}"
+  readlink -f "${jdk_source}"
+  rm -rf "${jdk_name}"
+  cp -rL "${jdk_source}" "${jdk_name}"
+}
+
+function unpackJdk() {
+  local jdk_source="${1}"
+  local jdk_name="${2}"
+  echo "${jdk_source} is file, will be unpacked and moved *as* ${jdk_name}"
+  readlink -f "${jdk_source}"
+  ls > before
+  if echo "${jdk_source}" | grep "\.zip" ; then
+    unzip "${jdk_source}"
+  else
+    tar -xf "${jdk_source}"
+  fi
+  ls > after
+  unpacked=$(getDiff)
+  local unpacked=${unpacked}
+  rm -rf "${jdk_name}"
+  mv "${unpacked}" "${jdk_name}"
+}
+
+function prepareAlternateJavaHome() {
+  local tag_os_arch="${1}"
+  local existing_archive="${2}"
+  futur_java_home="future_boot_jdk"
+  if [ -e "${futur_java_home}" ] ; then
+    rm -rf "${futur_java_home}"
+  fi
+  if [ -f "${existing_archive}" ] ; then
+    echo "${tag_os_arch} is there as ${existing_archive}, will be reused"
+    unpackJdk "${existing_archive}" "${futur_java_home}"
+  else
+    echo "${tag_os_arch} is tag:os.arch, will be downloaded and unpacked"
+    downloadAdoptiumReleases "${tag_os_arch}"
+    unpackJdk "${LAST_DOWNLOADED_FILE}" "${futur_java_home}"
+  fi
+  if uname | grep CYGWIN ; then
+    chmod777 "${futur_java_home}"
+  fi
+  JAVA_HOME=$(readlink -f "${futur_java_home}");
+  export JAVA_HOME="${JAVA_HOME}"
+}
+
+function tapsAndJunits() {
```

Comment:
This looks Red Hat specific? I'd rather this script be generic for comparing ie. you pass in "Version String's" etc... only

Reply:
Not exactly. It is generating junit-comaptible xml  and tap file to make automated processing slightly easier
I can move it else where, or make more easily on/off.

Reply:
Or maybe you think different part of RH specific? Can you be a bit more specific if so, please?

Reply:
I got a bit lost in the detail, so wasn't sure what it was doing, I guess it's test tooling compatible stuff, so that seems reasonable